amgut/lib/data_access/tests/test_ag_data_access.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_create_AGDataAccess(self):
+        raise NotImplementedError()
+
+    def test_delete_AGDataAccess(self):
+        raise NotImplementedError()
+
+    def test_get_cursor(self):
+        raise NotImplementedError()
+
+    def test_open_connection(self):
+        raise NotImplementedError()
+
+    def test_get_col_names_from_cursor(self):
+        raise NotImplementedError()
+
+    def test_authenticateWebAppUser(self):
+        raise NotImplementedError()
+
+    def test_addAGLogin(self):
+        raise NotImplementedError()
+
+    def test_getAGBarcodeDetails(self):
+        raise NotImplementedError()
+
+    def test_getAGKitDetails(self):
+        raise NotImplementedError()
+
+    def test_registerHandoutKit(self):
+        raise NotImplementedError()
+
+    def test_deleteAGParticipantSurvey(self):
+        raise NotImplementedError()
+
+    def test_getConsent(self):
+        raise NotImplementedError()
+
+    def test_logParticipantSample(self):
+        raise NotImplementedError()
+
+    def test_deleteSample(self):
+        raise NotImplementedError()
+
+    def test_getHumanParticipants(self):
+        raise NotImplementedError()
+
+    def test_is_old_survey(self):
+        raise NotImplementedError()
+
+    def test_updateVioscreenStatus(self):
+        raise NotImplementedError()
+
+    def test_getAnimalParticipants(self):
+        raise NotImplementedError()
+
+    def test_getParticipantSamples(self):
+        raise NotImplementedError()
+
+    def test_getEnvironmentalSamples(self):
+        raise NotImplementedError()
+
+    def test_getAvailableBarcodes(self):
+        raise NotImplementedError()
+
+    def test_verifyKit(self):
+        raise NotImplementedError()
+
+    def test_getMapMarkers(self):
+        raise NotImplementedError()
+
+    def test_handoutCheck(self):
+        raise NotImplementedError()
+
+    def test_check_access(self):
+        raise NotImplementedError()
+
+    def test_getAGKitIDsByEmail(self):
+        raise NotImplementedError()
+
+    def test_ag_set_pass_change_code(self):
+        raise NotImplementedError()
+
+    def test_ag_update_kit_password(self):
+        skid = 'test'
+        # Check and set pass_reset_code if needed
+        sql = '''UPDATE ag.ag_kit
+                 SET pass_reset_code = %s
+                 WHERE supplied_kit_id = %s AND pass_reset_code IS NULL
+                 RETURNING pass_reset_code'''
+        re_code = self.conn_handler.execute_fetchone(sql, ['T3ST', skid])[0]
+        # Verify password is not what we are changing it to
+        sql = 'SELECT kit_password FROM ag.ag_kit WHERE supplied_kit_id = %s'
+        old_pass = self.conn_handler.execute_fetchone(sql, [skid])[0]
+        self.assertFalse(bcrypt.verify('password', old_pass))
+
+        # Verify password change
+        test_sql = '''SELECT kit_password, pass_reset_code
+                      FROM ag.ag_kit
+                      WHERE supplied_kit_id = %s'''
+        try:
```

Comment:
I just looked at the qiita database test suite.

I haven't found a ___single___ try except statement in entire qiita database test suite.  In unittests, you want to raise exceptions, not catch them.  If you start adding try, except statements in your test cases, you are asking for poor test coverage.

Why can't we do the same thing that qiita does?  I vote to spend some proper time to thoroughly work through these test cases.  Rushing through them will seriously bite us in the butt later.  @wasade , @adamrp , @josenavas , @antgonza , @ElDeveloper , could you comment here to make sure that I'm not missing something.